lib/semantics/semantics.cc

```diff
@@ -14,6 +14,7 @@
 
 #include "semantics.h"
 #include "assignment.h"
+#include "statements.h"
```

Comment:
Would you consider renaming to check-error-stop? And renaming the class.

Reply:
I agree, that would be more consistent with the existing names. I think `check-stop.cc` would be fine since it checks STOP and ERROR STOP.

Reply:
Agreed with Tim.

Reply:
Done.